### PR TITLE
Collect build timings on GitHub CI

### PR DESCRIPTION
**Pull request by mrobinson**
This will help to investigate slow builds on GitHub.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they do not change behavior.

### CODE REVIEW

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -245,24 +245,29 @@ def test_unit(self, test_name=None, package=None, bench=False, nocapture=False,
...
-                                                    with_layout_2020=with_layout_2020,
-                                                    **kwargs)
-            if err:
-                return err
+        # Return if there is nothing to do.
+        if len(packages) == 0 and len(in_crate_packages) == 0:
+            return 0
+
+        # https://internals.rust-lang.org/t/exploring-crate-graph-build-times-with-cargo-build-ztimings/10975
+        # Prepend so that e.g. `-Ztimings` (which means `-Ztimings=info,html`)
```
Comment (atbrakhi) [inline python/servo/testing_commands.py:253]: did you mean to say `-Ztimings=info.html` ?

**Review by atbrakhi — APPROVED** [review]

File: python/servo/testing_commands.py (hunk 1 above)
Comment (mrobinson) [reply python/servo/testing_commands.py:253]: Oh, I copied the comment from the same lines that are used to set this up for `./mach build`, but the comment is seriously out of date. I'll update it and the old one as well.

**Comment by CYBAI** [comment]
Failed in Windows

```
Copy-Item: D:\a\_temp\155c05b6-5ff0-4c47-9a57-38b029b3efd9.ps1:2
Line |
   2 |  cp -r target/cargo-timings target/cargo-timings-windows-layout-20[13](https://github.com/servo/servo/actions/runs/5349061154/jobs/9699756977#step:10:14)
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path 'D:\a\servo\servo\target\cargo-timings' because it does not exist.
```

**Comment by mrobinson** [comment]
@bors-servo r=atbrakhi
 - Converted script in `windows.yml` to PowerShell

**Pull request merged**